loki/transformations/sanitise/tests/test_associates.py

```diff
@@ -148,6 +148,57 @@ def test_transform_associates_array_call(frontend):
     assert routine.variable_map['local_arr'].type.shape == ('some_obj%a%n',)
 
 
+@pytest.mark.parametrize('frontend', available_frontends(
+    skip=[(OMNI, 'OMNI does not handle missing type definitions')]
+))
+def test_transform_associates_array_slices(frontend):
+    """
+    Test the resolution of associated array slices.
+    """
+    fcode = """
+subroutine transform_associates_slices(arr2d)
+  use some_module, only: some_obj, another_routine
+  implicit none
+  real, intent(inout) :: arr2d(:,:)
+  integer :: i
+  integer, parameter :: idx_a = 2
+
+  associate (a => arr2d(:, 1), b=>arr2d(:, idx_a) )
+    b(:) = 42.0
+    do i=1, 5
+      a(i) = b(i+2)
+      call another_routine(i, a(2:4), b)
+    end do
+  end associate
```

Comment:
Do we also have a test that covers the reverse situation, where a dimension expression contains an associated symbol but the array is defined outside?

Something like:
```
real, intent(inout) :: arr2d(:,:)

associate(var_idx=>some%type%idx)
  do i=1,5
    arr2d(i, var_idx) = 1.0
  end do
end associate
```

Reply:
Yes, agreed, good point! I've extended the array slicing test to cover this example.